### PR TITLE
Doxygen comments for files in concurrency

### PR DESCRIPTION
**Pull request by aaron-tian**

### CODE REVIEW

**Comment by mengranwo** [comment]
@pervazea 
Can you take a look at this PR and request change if necessary?
@aaron-tian 
Also, can you add some simple description for what this PR is trying to do?

Thanks

**Comment by aaron-tian** [comment]
@mengranwo As title, add Doxygen comments for the files in the concurrency folder.

File: src/include/concurrency/timestamp_ordering_transaction_manager.h [hunk 1]
```diff
@@ -25,102 +25,273 @@ namespace concurrency {
...
+   * @return     True if owner, False otherwise.
+   */
   virtual bool IsOwned(
       TransactionContext *const current_txn,
       const storage::TileGroupHeader *const tile_group_header,
       const oid_t &tuple_id);
 
-  // This method tests whether the current transaction has created this version of the tuple
+  /**
+  * This method tests whether the current transaction has created this version of
```
Comment (pervazea) [inline src/include/concurrency/timestamp_ordering_transaction_manager.h]: This method tests -> Test

File: src/include/concurrency/timestamp_ordering_transaction_manager.h [hunk 2]
```diff
@@ -25,102 +25,273 @@ namespace concurrency {
...
+  * @return     True if written, False otherwise.
+  */
   virtual bool IsWritten(
       TransactionContext *const current_txn,
       const storage::TileGroupHeader *const tile_group_header,
       const oid_t &tuple_id);
 
-  // This method tests whether it is possible to obtain the ownership.
+  /**
+   * This method tests whether it is possible to obtain the ownership.
```
Comment (pervazea) [inline src/include/concurrency/timestamp_ordering_transaction_manager.h]: Test if this transaction can get ownership.

File: src/include/concurrency/timestamp_ordering_transaction_manager.h [hunk 3]
```diff
@@ -25,102 +25,273 @@ namespace concurrency {
...
   virtual void PerformUpdate(TransactionContext *const current_txn,
                              const ItemPointer &old_location,
                              const ItemPointer &new_location);
 
+  /**
+   * @brief      Perform a delete operation
+   *
+   * @param      current_txn   The current transaction
+   * @param[in]  old_location  The location of the old tuple to be deleted
+   * @param[in]  new_location  The location of the new tuple
```
Comment (pervazea) [inline src/include/concurrency/timestamp_ordering_transaction_manager.h]: Explain when this PerformDelete is used vs the one below?

File: src/include/concurrency/timestamp_ordering_transaction_manager.h [hunk 4]
```diff
@@ -25,102 +25,273 @@ namespace concurrency {
...
+   */
   bool SetLastReaderCommitId(
       const storage::TileGroupHeader *const tile_group_header,
       const oid_t &tuple_id, 
       const cid_t &current_cid, 
       const bool is_owner);
 
-  // Initiate reserved area of a tuple
+  /**
+   * Initiate reserved area of a tuple.
```
Comment (pervazea) [inline src/include/concurrency/timestamp_ordering_transaction_manager.h]: Initiate -> Initialize

File: src/include/concurrency/transaction_context.h [hunk 5]
```diff
@@ -128,78 +207,135 @@ class TransactionContext : public Printable {
...
   //===--------------------------------------------------------------------===//
   // Data members
   //===--------------------------------------------------------------------===//
 
-  // transaction id
+  /** transaction id */
   txn_id_t txn_id_;
 
-  // thread id
+  /** thread id */
```
Comment (pervazea) [inline src/include/concurrency/transaction_context.h]: id of thread creating this transaction?

File: src/include/concurrency/transaction_manager.h [hunk 6]
```diff
@@ -54,56 +66,128 @@ class TransactionManager {
...
+   */
   VisibilityType IsVisible(
       TransactionContext *const current_txn,
       const storage::TileGroupHeader *const tile_group_header,
       const oid_t &tuple_id,
       const VisibilityIdType type = VisibilityIdType::READ_ID);
 
-  // This method test whether the current transaction is the owner of this version.
+  /**
+   * This method test whether the current transaction is the owner of this
```
Comment (pervazea) [inline src/include/concurrency/transaction_manager.h]: test -> tests

File: src/include/concurrency/transaction_manager.h [hunk 7]
```diff
@@ -54,56 +66,128 @@ class TransactionManager {
...
+   * version.
```
Comment (pervazea) [inline src/include/concurrency/transaction_manager.h]: version -> tuple?

File: src/include/concurrency/transaction_manager.h [hunk 8]
```diff
@@ -54,56 +66,128 @@ class TransactionManager {
...
+   * @return     True if written, False otherwise.
+   */
   virtual bool IsWritten(
     TransactionContext *const current_txn,
     const storage::TileGroupHeader *const tile_group_header,
     const oid_t &tuple_id) = 0;
 
-  // This method tests whether it is possible to obtain the ownership.
+  /**
+   * This method tests whether it is possible to obtain the ownership.
```
Comment (pervazea) [inline src/include/concurrency/transaction_manager.h]: omit the.

File: src/include/concurrency/transaction_manager.h [hunk 9]
```diff
@@ -54,56 +66,128 @@ class TransactionManager {
...
+   * @return     True if ownable, False otherwise.
+   */
   virtual bool IsOwnable(
       TransactionContext *const current_txn,
       const storage::TileGroupHeader *const tile_group_header,
       const oid_t &tuple_id) = 0;
 
-  // This method is used to acquire the ownership of a tuple for a transaction.
+  /**
+   * This method is used to acquire the ownership of a tuple for a transaction.
```
Comment (pervazea) [inline src/include/concurrency/transaction_manager.h]: omit the.

File: src/include/concurrency/transaction_manager.h [hunk 10]
```diff
@@ -54,56 +66,128 @@ class TransactionManager {
...
   virtual bool AcquireOwnership(
       TransactionContext *const current_txn,
       const storage::TileGroupHeader *const tile_group_header, 
       const oid_t &tuple_id) = 0;
 
-  // This method is used by executor to yield ownership after the acquired ownership.
+  /**
+   * This method is used by executor to yield ownership after the acquired
+   * ownership.
+   *
```
Comment (pervazea) [inline src/include/concurrency/transaction_manager.h]: Used by the executor to yield ownership after acquiring it.

**Comment by mengranwo** [comment]
@aaron-tian 
Aaron, can you make changes according to @pervazea 's comment?

Thanks!

**Comment by aaron-tian** [comment]
@mengranwo Thanks for poking me. Will do it in two days.

**Comment by aaron-tian** [comment]
@mengranwo Done.

**Pull request merged**